### PR TITLE
Move JWE for user secrets to Kaggle Authorization header instead of r…

### PR DESCRIPTION
**Pull request by vimota**
…equest body.

### CODE REVIEW

**Comment by rosbo** [comment]
Adding @Philmod as a reviewer since he is the owner of the R image.

**Comment by vimota** [comment]
> Could you also add a similar change to `kaggle/kaggle_bigquery.R`?

Done!

File: kaggle/kaggle_bigquery.R [hunk 1]
```diff
@@ -27,8 +27,15 @@ TokenBigQueryKernel <- R6::R6Class("TokenBigQueryKernel", inherit = Token2.0, li
     if (KAGGLE_USER_SECRETS_TOKEN == '') {
       stop("Expected KAGGLE_USER_SECRETS_TOKEN environment variable to be present.", call. = FALSE)
     }
-    request_body <- list(JWE = KAGGLE_USER_SECRETS_TOKEN, Target = 1)
-    response <- POST(paste0(KAGGLE_BASE_URL, GET_USER_SECRET_ENDPONT), body = request_body, encode = "json")
+    request_body <- list(Target = 1)
+    auth_header <- paste0("Bearer ", KAGGLE_USER_SECRETS_TOKEN)
+    headers <- add_headers(c("X-Kaggle-Authorization" = auth_header))
+    response <- POST(paste0(KAGGLE_BASE_URL, GET_USER_SECRET_ENDPONT),
```
Comment (Philmod) [inline kaggle/kaggle_bigquery.R:33]: nit: ENDPOINT (missing a I), which is also misspelled on line 13

File: kaggle/kaggle_secrets.R [hunk 2]
```diff
@@ -13,8 +13,17 @@ get_user_secret <- function(label) {
     if (KAGGLE_USER_SECRETS_TOKEN == '') {
       stop("Expected KAGGLE_USER_SECRETS_TOKEN environment variable to be present.", call. = FALSE)
     }
-    request_body <- list(JWE = KAGGLE_USER_SECRETS_TOKEN, Label = label)
-    response <- POST(paste0(KAGGLE_BASE_URL, GET_USER_SECRET_BY_LABEL_ENDPONT), body = request_body, encode = "json")
+    request_body <- list(Label = label)
+    auth_header <- paste0("Bearer ", KAGGLE_USER_SECRETS_TOKEN)
+    headers <- add_headers(c("X-Kaggle-Authorization" = auth_header))
+    response <- POST(
+      paste0(KAGGLE_BASE_URL,GET_USER_SECRET_BY_LABEL_ENDPONT),
```
Comment (Philmod) [inline kaggle/kaggle_secrets.R:20]: idem

Comment (Philmod) [reply kaggle/kaggle_secrets.R:20]: nit: missing a space after `,` to be consistent

File: kaggle/kaggle_bigquery.R (hunk 1 above)
Comment (vimota) [reply kaggle/kaggle_bigquery.R]: Done

File: kaggle/kaggle_secrets.R (hunk 2 above)
Comment (vimota) [reply kaggle/kaggle_secrets.R]: Done

**Comment by vimota** [comment]
> LGTM for a quick fix, but could we add a proper test in a separate PR?
> A unit test ([`master`/tests/test_bigquery.R](https://github.com/Kaggle/docker-rstats/blob/master/tests/test_bigquery.R)), and/or a release test (http://shortn/_AufTDnI8SA)?

Added the unit test and will add a release test in a separate PR, thanks!

**Pull request merged**